src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/BotActions/RestartGameBotActionData.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using RegressionGames.StateRecorder.JsonConverters;
+using RegressionGames.StateRecorder.Models;
+using StateRecorder.BotSegments.Models;
+// ReSharper disable once RedundantUsingDirective - used in #else.. don't remove
+using UnityEngine.SceneManagement;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+namespace RegressionGames.StateRecorder.BotSegments.Models.BotActions
+{
+    /**
+     * <summary>An action to restart the game.  This should only be used as the 'last' segment in your sequence or segment list until REG-2170.</summary>
+     */
+    [Serializable]
+    public class RestartGameBotActionData : IBotActionData
+    {
+        [NonSerialized]
+        public static readonly BotActionType Type = BotActionType.RestartGame;
+
+        private static IRGRestartGameAction _action;
+
+        public int apiVersion = SdkApiVersion.VERSION_25;
+
+        [NonSerialized]
+        private bool _isStopped;
+
+        private string _error;
+
+        public void StartAction(int segmentNumber, Dictionary<long, ObjectStatus> currentTransforms, Dictionary<long, ObjectStatus> currentEntities)
+        {
+            if (!_isStopped)
+            {
+                if (_action == null)
+                {
+#if UNITY_EDITOR
+                    // no-op
+#else
+
+                    // load the type on another thread to avoid 'hitching' the game
+                    new Thread(() =>
+                    {
+                        // load our script type without knowing the assembly name, just the full type
+                        var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+                        var allAssembliesLength = allAssemblies.Length;
+                        for (var i = 0; _action == null && i < allAssembliesLength; i++)
+                        {
+                            var a = allAssemblies[i];
+                            foreach (var type in a.GetTypes())
+                            {
+                                if (typeof(IRGRestartGameAction).IsAssignableFrom(type))
+                                {
+                                    RGDebug.LogInfo($"Using the '{type.FullName}' implementation of IRGRestartGameAction to process Bot Restart Game Actions.  If this is not the type you expected to handle this action, you may accidentally have more than one implementation of the IRGRestartGameAction interface in your runtime.");
+                                    _action = (IRGRestartGameAction)Activator.CreateInstance(type);
+                                }
+                                break;
```

Comment:
do we want this `break` inside the `if` above it?

Reply:
ha.. yeah probably